PluralKit.Bot/Commands/Message.cs

```diff
@@ -22,23 +24,70 @@ public class ProxiedMessage
     private static readonly Duration EditTimeout = Duration.FromMinutes(10);
 
     // private readonly IDiscordCache _cache;
+    private readonly ModelRepository _repo;
     private readonly IClock _clock;
+    private readonly IMetrics _metrics;
 
     private readonly EmbedService _embeds;
     private readonly LogChannelService _logChannel;
     private readonly DiscordApiClient _rest;
     private readonly WebhookExecutorService _webhookExecutor;
+    private readonly ProxyService _proxy;
 
     public ProxiedMessage(EmbedService embeds, IClock clock,
-                          DiscordApiClient rest,
+                          DiscordApiClient rest, IMetrics metrics, ModelRepository repo, ProxyService proxy,
                           WebhookExecutorService webhookExecutor, LogChannelService logChannel, IDiscordCache cache)
     {
         _embeds = embeds;
         _clock = clock;
         _rest = rest;
         _webhookExecutor = webhookExecutor;
+        _repo = repo;
         _logChannel = logChannel;
         // _cache = cache;
+        _metrics = metrics;
+        _proxy =  proxy;
+    }
+
+    public async Task ReproxyMessage(Context ctx)
+    {
+        var msg = await GetMessageToEdit(ctx);
+
+        if (ctx.System.Id != msg.System?.Id)
+            throw new PKError("Can't reproxy a message sent by a different system.");
+
+        // Bail if the message is more than 2 minutes old
+        var msgTimestamp = Instant.FromUnixTimeMilliseconds((long)(msg.Message.Mid >> 22) + 1420070400000);
+        if (msgTimestamp.Plus(Duration.FromSeconds(60 * 2)) < SystemClock.Instance.GetCurrentInstant())
+            throw new PKError("The message is too old to be reproxied.");
+
+        // Get target member ID
+        var target = await ctx.MatchMember();
```

Comment:
I think most people expect this command to work with proxy tags instead of a member name/id

Reply:
problem with using proxy tags here is that we then have no easy way to distinguish between a prefix and a suffix. if proxy tags were prefix only (or suffix only)  then that would work, but they're not - so you'd have to parse them like the proxy tag set method does (with the `text` sentinel). looking up a member by name/ID makes this a lot easier both in implementation and in usage

Reply:
I was going to suggest using a ProxyTagParser then I remembered this method doesn't actually take any content. makes sense to use a member name/id then